src/pretalx/agenda/views/schedule.py

```diff
@@ -65,6 +68,46 @@ def dispatch(self, request, *args, **kwargs):
             )
         return super().dispatch(request, *args, **kwargs)
 
+    @staticmethod
+    def generate_ics_token(user_id):
+        """Generate a signed token with user ID and 15-day expiry"""
+        expiry = timezone.now() + timedelta(days=15)
+        value = {"user_id": user_id, "exp": int(expiry.timestamp())}
+        return signing.dumps(value, salt="my-starred-ics")
+
+    @staticmethod
+    def parse_ics_token(token):
+        """Parse and validate the token, return user_id if valid"""
+        try:
+            value = signing.loads(token, salt="my-starred-ics", max_age=15*24*60*60)
+            if value["exp"] < int(timezone.now().timestamp()):
+                raise ValueError("Token expired")
+            return value["user_id"]
+        except (signing.BadSignature, signing.SignatureExpired, KeyError, ValueError) as e:
+            logger.warning(f"Failed to parse ICS token: {e}")
```

Comment:
Wrong usage of logging. Do:

```py
logger.warning('Failed to parse ICS token: %s', e)
```

The reason is that, the point of using logging is to "not building message when it is not necessary". If the application log level is `ERROR`, all log messages less severe than `ERROR` will be skipped.
 Your usage makes the log message to be always built, it is wasteful.
The suggested usage will allow to delay the building until it is a condition met.